File: database/src/main/java/org/difft/app/database/models/SharedContactModel.java
package org.difft.app.database.models;

import com.tencent.wcdb.WCDBField;
import com.tencent.wcdb.WCDBIndex;
import com.tencent.wcdb.WCDBTableCoding;

import java.util.Objects;

@WCDBTableCoding
public class SharedContactModel {
    @WCDBField(isPrimary = true, isAutoIncrement = true)
    public long databaseId;

    @WCDBIndex
    @WCDBField
    public String messageId;

    @WCDBField
    public String givenName;

    @WCDBField
    public String familyName;

    @WCDBField
    public String namePrefix;

    @WCDBField
    public String nameSuffix;

    @WCDBField
    public String middleName;

    @WCDBField
    public String displayName;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        SharedContactModel that = (SharedContactModel) o;
        return Objects.equals(messageId, that.messageId) && Objects.equals(givenName, that.givenName) && Objects.equals(familyName, that.familyName) && Objects.equals(namePrefix, that.namePrefix) && Objects.equals(nameSuffix, that.nameSuffix) && Objects.equals(middleName, that.middleName) && Objects.equals(displayName, that.displayName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(messageId, givenName, familyName, namePrefix, nameSuffix, middleName, displayName);
    }
}

File: database/src/main/java/org/difft/app/database/models/SharedContactModel.kt
package org.difft.app.database.models

import com.tencent.wcdb.WCDBField
import com.tencent.wcdb.WCDBIndex
import com.tencent.wcdb.WCDBTableCoding
import java.util.Objects

@WCDBTableCoding
class SharedContactModel {
    // Int (not Long): WCDB-KSP generates `== 0` autoincrement guard which does not
    // compile for Long primary keys. Column type stays ColumnType.Integer — no schema
    // change. Local rowid never exceeds 2^31 and is never externally serialized. #901
    @WCDBField(isPrimary = true, isAutoIncrement = true)
    var databaseId: Int = 0

    @WCDBIndex
    @WCDBField
    var messageId: String? = null

    @WCDBField
    var givenName: String? = null

    @WCDBField
    var familyName: String? = null

    @WCDBField
    var namePrefix: String? = null

    @WCDBField
    var nameSuffix: String? = null

    @WCDBField
    var middleName: String? = null

    @WCDBField
    var displayName: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        other as SharedContactModel
        return messageId == other.messageId &&
                givenName == other.givenName &&
                familyName == other.familyName &&
                namePrefix == other.namePrefix &&
                nameSuffix == other.nameSuffix &&
                middleName == other.middleName &&
                displayName == other.displayName
    }

    override fun hashCode(): Int =
        Objects.hash(messageId, givenName, familyName, namePrefix, nameSuffix, middleName, displayName)
}
